browser/src/components/core/Editor.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { useCallback, useState } from "react";
+import Modal from "react-modal";
+// import { parse } from "node-html-parser";
+
+// Tiptap + extensions
+import { useEditor, EditorContent } from "@tiptap/react";
+import Document from "@tiptap/extension-document";
+import Paragraph from "@tiptap/extension-paragraph";
+import Text from "@tiptap/extension-text";
+import Bold from "@tiptap/extension-bold";
+import Italic from "@tiptap/extension-italic";
+import Link from "@tiptap/extension-link";
+import Placeholder from "@tiptap/extension-placeholder";
+import History from "@tiptap/extension-history";
+
+import { ButtonClass } from "src/types/styles";
+import "./Editor.css";
+
+interface Props {
+  value?: string;
+  onChange: (value: string) => void;
+  placeholder?: string;
+}
+
+Modal.setAppElement("#root");
+export function Editor({
+  value,
+  onChange,
+  placeholder,
+}: Props) {
+  const [linkInput, setLinkInput] = useState<string|null>(null);
+  const [displayLinkModal, setDisplayLinkModal] = useState<boolean>(false);
+  
+  const sanitize = (inputHtml: string): string => {
+    // Remove first p tag to prevent text going to next line
+    return inputHtml.replace(
+      /<p[^>]*>|<\/p[^>]*>/,
+      ""
+    );
+
+    //var root = parse(inputHtml);
+    //var result = ""
+    //// Remove first p tag
+    //if (root.firstChild.tagName == "P") { 
+    //  result += root.firstChild.text;  
+    //}
+
+    //// root = root.firstChild
+    // const tagAllowlist = new Set(["P", "STRONG", "EM", "A"]);
+    // while(root.firstChild.firstChild != null) {
+    //   console.log(root.firstChild)
+    //   console.log(root.text)
+    //   const child = root.firstChild; 
+    //   if (!(tagAllowlist.has(child.tagName))) {
+    //     result += child.text;
+    //   } else {
+    //     result += child;
+    //   }
+    //   root = root.firstChild;
+
+    //   //else if (child.tagName == "a") {
+    //     // Remove all attributes of a tag except href, target, and rel
+    //   //}
+    // }
+    // return result;
+    //console.log(result)
+  };
+
+  const openModal = () => {
+    setDisplayLinkModal(true)
+    toggleBackgroundScrollingOnModal(false);
+  }
+
+  // Set Cmd/Ctrl-k shortcut
+  const CustomLink = Link.extend({
+    addKeyboardShortcuts() {
+      return {
+        'Mod-k': () => openModal(),
+      }
+    },
+  })
+
+  const editor = useEditor({
+    extensions: [
+      Document,
+      Paragraph,
+      Text,
+      History,
+      Bold,
+      Italic,
+      CustomLink,
+      Placeholder.configure({
+        placeholder: `${placeholder}`,
+      }),
+    ],
+    onUpdate: ({ editor }) => {
+      onChange(
+        sanitize(
+          editor.getHTML()
```

Comment:
Another option here if we're worried is to use `getJSON` instead, which exports in tiptap's JSON schema, and then sanitize by looping through and making sure there are only `em`, `strong`, `p`, and `a` tags, and then calling tiptap's `generateHTML` on the contribution card to to render.

Reply:
although I've tested with writing `<script>` tags, and Tiptap doesn't serialize it and instead renders as plain text.